File: app/src/main/java/com/botijasoftware/utils/interpolator/ParabolicInterpolator.java
package com.botijasoftware.utils.interpolator;

public class ParabolicInterpolator implements Interpolator {
	//public abstract Interpolator() {}
    private float amount;

    public ParabolicInterpolator(float amount) {
        this.amount = amount;
    }
	public float interpolate(float v0, float v1, float t) {

		return v0 + t*(v1 - v0) +  amount * (float) Math.sin(Math.PI * t)*(v1 - v0);

	}
}

File: app/src/main/java/com/botijasoftware/utils/interpolator/ParabolicInterpolator.kt
package com.botijasoftware.utils.interpolator

class ParabolicInterpolator(//public abstract Interpolator() {}
        private val amount: Float) : Interpolator {
    override fun interpolate(v0: Float, v1: Float, t: Float): Float {

        return v0 + t * (v1 - v0) + amount * Math.sin(Math.PI * t).toFloat() * (v1 - v0)

    }
}
